### ISSUE TITLE
PyPy cpyext: docstrings of `getset` entries not correctly propagated

### ISSUE THREAD

**Issue by wjakob** [pypy/pypy#4898]
Dear @mattip,

I encountered the following issue with PyPy's cpyext: If you register a custom type with `tp_getset`, and if that list of `PyGetSetDef ` entries contains docstrings, then these are lost.
```
static PyGetSetDef test_getset[] = {
    { "prop", test_getter, NULL, "A docstring", NULL },
    {NULL, NULL, NULL, NULL, NULL }
};
```

Here is a simple reproducer: https://github.com/wjakob/pypy_issue_getset_doc

If you compile and install it (`pypy -m pip install .`) and run ``issue.py``, which contains
```python
from pypy_issues import Test
assert Test.prop.__doc__ == 'A docstring'
```
this will fail with
```
Traceback (most recent call last):
  File "/Users/wjakob/pypy_issues/issue.py", line 2, in <module>
    assert Test.prop.__doc__ == 'A docstring'
AssertionError
```
On CPython, the docstring is set, and there is no assertion failure.

(Tested on `pypy3.10-v7.3.15-macos_arm64`)

**Comment by mattip** [comment]
Maybe connected to #3742. Thanks for the clear reproducer, it will make the problem easier to fix.

**Comment by mattip** [comment]
Fixed in #4904
